src/state/opamEnv.ml

```diff
@@ -874,35 +874,32 @@ let env_hook_script shell =
     (env_hook_script_base shell)
 
 let source root shell f =
-  let fname = OpamFilename.prettify (OpamPath.init root // f) in
-  let unix_transform ?using_backslashes () =
-    let cygpath = Lazy.force OpamSystem.get_cygpath_path_transform in
-    cygpath ~pathlist:false fname
-    |> OpamStd.Env.escape_single_quotes ?using_backslashes
+  let if_exists_source: _ format = match shell with
+    | SH_csh -> "if ( -f '%s' ) source '%s'\n"
+    | SH_fish -> "test -r '%s' && source '%s'\n"
+    | SH_sh | SH_bash -> "test -r '%s' && . '%s'\n"
+    | SH_zsh -> "[[ -r '%s' ]] && source '%s'\n"
+    | SH_cmd -> "if exist \"%s\" call \"%s\"\n"
+    | SH_pwsh _ -> "if Test-Path \"%s\" { . \"%s\" }\n"
```

Comment:
The semantics of "an error can lead to terminating the calling script at an early stage" are incredibly finnicky. See for example [this discussion on bash wiki](https://mywiki.wooledge.org/BashFAQ/105). I have found on my system (with bash 5.1.16, because apparently that's relevant) that even with `set -eu` at the start of `.bashrc` (which no one should ever do because that can brick your system), a line `test -r "doesnt_exist" && source "doesnt_exist"` does not interrupt the program.

Reply:
In my shell (bash, 5.2.21), having `set -e ; test -d nosuchdir && echo no dir;` in my bashrc makes me no more able to open a new (bash) terminal.

Reply:
holy shit bash is insane
alright let's put it back in :)